packages/apps/reputation-oracle/server/src/modules/qualification/qualification.service.ts

```diff
@@ -0,0 +1,189 @@
+import { HttpStatus, Injectable, Logger } from '@nestjs/common';
+import {
+  CreateQualificationDto,
+  AssignQualificationDto,
+  UnassignQualificationDto,
+  QualificationDto,
+} from './qualification.dto';
+import { QualificationEntity } from './qualification.entity';
+import { ErrorQualification } from '../../common/constants/errors';
+import { ControlledError } from '../../common/errors/controlled';
+import { QualificationRepository } from './qualification.repository';
+import { UserEntity } from '../user/user.entity';
+import { UserRepository } from '../user/user.repository';
+import { UserStatus, Role } from '../../common/enums/user';
+
+@Injectable()
+export class QualificationService {
+  private readonly logger = new Logger(QualificationService.name);
+
+  constructor(
+    private readonly qualificationRepository: QualificationRepository,
+    private readonly userRepository: UserRepository,
+  ) {}
+
+  public async createQualification(
+    createQualificationDto: CreateQualificationDto,
+  ): Promise<QualificationDto> {
+    const newQualification = new QualificationEntity();
+    newQualification.reference = createQualificationDto.reference;
+    newQualification.title = createQualificationDto.title;
+    newQualification.description = createQualificationDto.description;
+
+    if (createQualificationDto.expiresAt) {
+      const providedExpiresAt = new Date(createQualificationDto.expiresAt);
+
+      if (providedExpiresAt <= new Date()) {
```

Comment:
Maybe lets add some reasonable delta here?
For example, qualification should be valid for at least an hour or a day.